diff --git a/libpiper/CMakeLists.txt b/libpiper/CMakeLists.txt
index 57349c1..74f0140 100644
--- a/libpiper/CMakeLists.txt
+++ b/libpiper/CMakeLists.txt
@@ -8,23 +8,44 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 include(ExternalProject)
 
 # Install location for espeak-ng
-set(ESPEAKNG_BUILD_DIR ${CMAKE_BINARY_DIR}/espeak_ng)
-set(ESPEAKNG_INSTALL_DIR ${CMAKE_BINARY_DIR}/espeak_ng-install)
+set(ESPEAKNG_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/espeak_ng)
+set(ESPEAKNG_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/espeak_ng-install)
 
 if(WIN32)
     # Special handling for Windows
     set(ESPEAKNG_STATIC_LIB ${ESPEAKNG_INSTALL_DIR}/lib/espeak-ng.lib)
-    set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/ucd.lib)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/${CMAKE_BUILD_TYPE}/ucd.lib)
+    else()
+        set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/ucd.lib)
+    endif()
 else()
     set(ESPEAKNG_STATIC_LIB ${ESPEAKNG_INSTALL_DIR}/lib/libespeak-ng.a)
     set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/libucd.a)
 endif()
 
+if(WIN32)
+    find_package(Git REQUIRED)
+    set(ESPEAKNG_PATCH_MSG "Patching espeak-ng for fixing Windows build")
+    set(ESPEAKNG_PATCH_CMD ${GIT_EXECUTABLE} apply ${CMAKE_CURRENT_LIST_DIR}/cmake/patches/espeak-ng-static-win.patch)
+endif()
+
+find_package(Git REQUIRED)
+
 ExternalProject_Add(espeak_ng_external
     GIT_REPOSITORY https://github.com/espeak-ng/espeak-ng.git
     GIT_TAG 212928b394a96e8fd2096616bfd54e17845c48f6  # 2025-Mar-22
     PREFIX ${ESPEAKNG_BUILD_DIR}
+
+    PATCH_COMMAND
+        ${CMAKE_COMMAND} -E echo ${ESPEAKNG_PATCH_MSG}
+        COMMAND ${ESPEAKNG_PATCH_CMD}
+
+    CMAKE_GENERATOR ${CMAKE_GENERATOR}
     CMAKE_ARGS
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
         -DCMAKE_INSTALL_PREFIX=${ESPEAKNG_INSTALL_DIR}
         -DBUILD_SHARED_LIBS:BOOL=OFF
         -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
@@ -37,8 +58,8 @@ ExternalProject_Add(espeak_ng_external
         -DEXTRA_cmn:BOOL=ON
         -DEXTRA_ru:BOOL=ON
         # Need to explicitly add ucd include directory for CI
-        "-DCMAKE_C_FLAGS=-D_FILE_OFFSET_BITS=64 -I${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external/src/ucd-tools/src/include"
-        "-DCMAKE_CXX_FLAGS=-D_FILE_OFFSET_BITS=64 -I${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external/src/ucd-tools/src/include"
+        "-DCMAKE_C_FLAGS=-D_FILE_OFFSET_BITS=64 -DESPEAK_NG_STATIC -I${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external/src/ucd-tools/src/include"
+        "-DCMAKE_CXX_FLAGS=-D_FILE_OFFSET_BITS=64 -DESPEAK_NG_STATIC -I${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external/src/ucd-tools/src/include"
     BUILD_BYPRODUCTS
         ${ESPEAKNG_STATIC_LIB}
         ${UCD_STATIC_LIB}
@@ -116,6 +137,10 @@ add_library(piper SHARED
     ${CMAKE_CURRENT_SOURCE_DIR}/src/piper.cpp
 )
 
+if(LIBPIPER_EXPORT_SYMBOLS)
+    target_compile_definitions(piper PRIVATE BUILDING_LIBPIPER)
+endif()
+
 add_dependencies(piper espeak_ng_external)
 target_include_directories(piper PUBLIC
     ${ESPEAKNG_INSTALL_DIR}/include
