diff --git a/libpiper/include/piper.h b/libpiper/include/piper.h
index 25841c2..d7ddc77 100644
--- a/libpiper/include/piper.h
+++ b/libpiper/include/piper.h
@@ -10,6 +10,16 @@
 extern "C" {
 #endif
 
+#if defined (_WIN32)
+  #if defined(BUILDING_LIBPIPER)
+    #define EXPORT_SYMBOL __declspec( dllexport )
+  #else
+    #define EXPORT_SYMBOL __declspec( dllimport )
+  #endif
+#else
+  #define EXPORT_SYMBOL
+#endif
+
 #define PIPER_OK 0
 #define PIPER_DONE 1
 #define PIPER_ERR_GENERIC -1
@@ -155,6 +165,7 @@ typedef struct piper_synthesize_options {
  *
  * \return a Piper text-to-speech synthesizer for the voice model.
  */
+EXPORT_SYMBOL
 piper_synthesizer *piper_create(const char *model_path, const char *config_path,
                                 const char *espeak_data_path);
 
@@ -163,6 +174,7 @@ piper_synthesizer *piper_create(const char *model_path, const char *config_path,
  *
  * \param synth Piper synthesizer.
  */
+EXPORT_SYMBOL
 void piper_free(piper_synthesizer *synth);
 
 /**
@@ -172,6 +184,7 @@ void piper_free(piper_synthesizer *synth);
  *
  * \return synthesis options from voice config.
  */
+EXPORT_SYMBOL
 piper_synthesize_options
 piper_default_synthesize_options(piper_synthesizer *synth);
 
@@ -188,6 +201,7 @@ piper_default_synthesize_options(piper_synthesizer *synth);
  *
  * \return PIPER_OK or error code.
  */
+EXPORT_SYMBOL
 int piper_synthesize_start(piper_synthesizer *synth, const char *text,
                            const piper_synthesize_options *options);
 
@@ -208,6 +222,7 @@ int piper_synthesize_start(piper_synthesizer *synth, const char *text,
  *
  * \return PIPER_DONE when complete, otherwise PIPER_OK or error code.
  */
+EXPORT_SYMBOL
 int piper_synthesize_next(piper_synthesizer *synth, piper_audio_chunk *chunk);
 
 #ifdef __cplusplus
