diff --git a/libpiper/src/piper.cpp b/libpiper/src/piper.cpp
index 774e87d..c54ab7f 100644
--- a/libpiper/src/piper.cpp
+++ b/libpiper/src/piper.cpp
@@ -96,8 +96,16 @@ struct piper_synthesizer *piper_create(const char *model_path,
     synth->session_options.DisableMemPattern();
     synth->session_options.DisableProfiling();

+#if defined(_WIN32) || defined(_WIN64)
+    std::string model_path_str(model_path);
+    synth->session = std::make_unique<Ort::Session>(
+        Ort::Session(ort_env,
+                     std::wstring(model_path_str.begin(), model_path_str.end()).c_str(),
+                     synth->session_options));
+#else
     synth->session = std::make_unique<Ort::Session>(
         Ort::Session(ort_env, model_path, synth->session_options));
+#endif

     return synth;
 }
