diff --git a/libpiper/CMakeLists.txt b/libpiper/CMakeLists.txt
index 57349c1..fd2be9a 100644
--- a/libpiper/CMakeLists.txt
+++ b/libpiper/CMakeLists.txt
@@ -8,23 +8,42 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 include(ExternalProject)
 
 # Install location for espeak-ng
-set(ESPEAKNG_BUILD_DIR ${CMAKE_BINARY_DIR}/espeak_ng)
-set(ESPEAKNG_INSTALL_DIR ${CMAKE_BINARY_DIR}/espeak_ng-install)
+set(ESPEAKNG_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/espeak_ng)
+set(ESPEAKNG_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/espeak_ng-install)
 
 if(WIN32)
     # Special handling for Windows
     set(ESPEAKNG_STATIC_LIB ${ESPEAKNG_INSTALL_DIR}/lib/espeak-ng.lib)
-    set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/ucd.lib)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/ucd.lib)
+    else()
+        set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/ucd.lib)
+    endif()
 else()
     set(ESPEAKNG_STATIC_LIB ${ESPEAKNG_INSTALL_DIR}/lib/libespeak-ng.a)
     set(UCD_STATIC_LIB ${ESPEAKNG_BUILD_DIR}/src/espeak_ng_external-build/src/ucd-tools/libucd.a)
 endif()
 
+if(WIN32)
+    find_package(Git REQUIRED)
+    set(ESPEAKNG_PATCH_MSG "Patching espeak-ng for fixing Windows build")
+    set(ESPEAKNG_PATCH_CMD ${GIT_EXECUTABLE} apply ${CMAKE_CURRENT_LIST_DIR}/cmake/patches/espeak-ng-static-win.patch)
+endif()
+
 ExternalProject_Add(espeak_ng_external
     GIT_REPOSITORY https://github.com/espeak-ng/espeak-ng.git
     GIT_TAG 212928b394a96e8fd2096616bfd54e17845c48f6  # 2025-Mar-22
     PREFIX ${ESPEAKNG_BUILD_DIR}
+
+    PATCH_COMMAND
+        ${CMAKE_COMMAND} -E echo ${ESPEAKNG_PATCH_CMD_MSG}
+        COMMAND ${ESPEAKNG_PATCH_CMD}
+ 
+    CMAKE_GENERATOR ${CMAKE_GENERATOR}
     CMAKE_ARGS
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
         -DCMAKE_INSTALL_PREFIX=${ESPEAKNG_INSTALL_DIR}
         -DBUILD_SHARED_LIBS:BOOL=OFF
         -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
@@ -115,6 +134,9 @@ endif()
 add_library(piper SHARED
     ${CMAKE_CURRENT_SOURCE_DIR}/src/piper.cpp
 )
+if(LIBPIPER_EXPORT_SYMBOLS)
+    target_compile_definitions(piper PRIVATE BUILDING_LIBPIPER)
+endif()
 
 add_dependencies(piper espeak_ng_external)
 target_include_directories(piper PUBLIC
diff --git a/libpiper/cmake/patches/espeak-ng-static-win.patch b/libpiper/cmake/patches/espeak-ng-static-win.patch
new file mode 100644
index 0000000..3d0c19b
--- /dev/null
+++ b/libpiper/cmake/patches/espeak-ng-static-win.patch
@@ -0,0 +1,60 @@
+diff --git a/src/include/espeak-ng/espeak_ng.h b/src/include/espeak-ng/espeak_ng.h
+index 81988e4..e4f068f 100644
+--- a/src/include/espeak-ng/espeak_ng.h
++++ b/src/include/espeak-ng/espeak_ng.h
+@@ -26,15 +26,7 @@ extern "C"
+ {
+ #endif
+ 
+-#if defined(_WIN32) || defined(_WIN64)
+-#ifdef LIBESPEAK_NG_EXPORT
+-#define ESPEAK_NG_API __declspec(dllexport)
+-#else
+-#define ESPEAK_NG_API __declspec(dllimport)
+-#endif
+-#else
+ #define ESPEAK_NG_API
+-#endif
+ 
+ #define ESPEAKNG_DEFAULT_VOICE "en"
+ 
+diff --git a/src/include/espeak-ng/speak_lib.h b/src/include/espeak-ng/speak_lib.h
+index 9ec4caf..63434e2 100644
+--- a/src/include/espeak-ng/speak_lib.h
++++ b/src/include/espeak-ng/speak_lib.h
+@@ -28,15 +28,13 @@
+ #include <stdio.h>
+ #include <stddef.h>
+ 
+-#if defined(_WIN32) || defined(_WIN64)
+-#ifdef LIBESPEAK_NG_EXPORT
+-#define ESPEAK_API __declspec(dllexport)
+-#else
+-#define ESPEAK_API __declspec(dllimport)
++#if defined(_WIN32)
++#include <io.h>
++#include <fcntl.h>
+ #endif
+-#else
++
+ #define ESPEAK_API
+-#endif
++
+ 
+ #define ESPEAK_API_REVISION  12
+ /*
+diff --git a/src/libespeak-ng/CMakeLists.txt b/src/libespeak-ng/CMakeLists.txt
+index bb39dc2..249e237 100644
+--- a/src/libespeak-ng/CMakeLists.txt
++++ b/src/libespeak-ng/CMakeLists.txt
+@@ -64,10 +64,6 @@ if (NOT MSVC)
+     "-Wmisleading-indentation"
+   )
+ endif()
+-target_compile_definitions(espeak-ng PRIVATE "LIBESPEAK_NG_EXPORT=1")
+-if (NOT BUILD_SHARED_LIBS)
+-  target_compile_definitions(espeak-ng INTERFACE "LIBESPEAK_NG_EXPORT=1")
+-endif()
+ 
+ target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=\"${CMAKE_INSTALL_PREFIX}/share/espeak-ng-data\"")
+ 
diff --git a/libpiper/include/piper.h b/libpiper/include/piper.h
index 25841c2..4d432c6 100644
--- a/libpiper/include/piper.h
+++ b/libpiper/include/piper.h
@@ -6,6 +6,16 @@
 #include <stdint.h>
 #include <uchar.h>
 
+#if defined (_WIN32)
+  #if defined(BUILDING_LIBPIPER)
+    #define EXPORT_SYMBOL __declspec( dllexport )
+  #else
+    #define EXPORT_SYMBOL __declspec( dllimport )
+  #endif
+#else
+  #define EXPORT_SYMBOL
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -155,6 +165,7 @@ typedef struct piper_synthesize_options {
  *
  * \return a Piper text-to-speech synthesizer for the voice model.
  */
+EXPORT_SYMBOL
 piper_synthesizer *piper_create(const char *model_path, const char *config_path,
                                 const char *espeak_data_path);
 
@@ -163,6 +174,7 @@ piper_synthesizer *piper_create(const char *model_path, const char *config_path,
  *
  * \param synth Piper synthesizer.
  */
+EXPORT_SYMBOL
 void piper_free(piper_synthesizer *synth);
 
 /**
@@ -172,6 +184,7 @@ void piper_free(piper_synthesizer *synth);
  *
  * \return synthesis options from voice config.
  */
+EXPORT_SYMBOL
 piper_synthesize_options
 piper_default_synthesize_options(piper_synthesizer *synth);
 
@@ -188,6 +201,7 @@ piper_default_synthesize_options(piper_synthesizer *synth);
  *
  * \return PIPER_OK or error code.
  */
+EXPORT_SYMBOL
 int piper_synthesize_start(piper_synthesizer *synth, const char *text,
                            const piper_synthesize_options *options);
 
@@ -208,6 +222,7 @@ int piper_synthesize_start(piper_synthesizer *synth, const char *text,
  *
  * \return PIPER_DONE when complete, otherwise PIPER_OK or error code.
  */
+EXPORT_SYMBOL
 int piper_synthesize_next(piper_synthesizer *synth, piper_audio_chunk *chunk);
 
 #ifdef __cplusplus
diff --git a/libpiper/src/piper.cpp b/libpiper/src/piper.cpp
index 774e87d..48abd82 100644
--- a/libpiper/src/piper.cpp
+++ b/libpiper/src/piper.cpp
@@ -96,8 +96,16 @@ struct piper_synthesizer *piper_create(const char *model_path,
     synth->session_options.DisableMemPattern();
     synth->session_options.DisableProfiling();
 
+#if defined(_WIN32) || defined(_WIN64)
+    auto model_path_strv = std::string_view(model_path);
+    synth->session = std::make_unique<Ort::Session>(
+        Ort::Session(ort_env,
+                    std::wstring(model_path_strv.begin(), model_path_strv.end()).c_str(),
+                    synth->session_options));
+#else
     synth->session = std::make_unique<Ort::Session>(
         Ort::Session(ort_env, model_path, synth->session_options));
+#endif
 
     return synth;
 }
